File: 02-programacion-basica/01-entorno-sintaxis-y-entrada-salida/java/ejemplos-guiados/HolaEntradaSalida.java
import java.util.Scanner;

public class HolaEntradaSalida {
    public static void main(String[] args) {
        // Scanner conecta el programa con la entrada estándar (teclado).
        try (Scanner scanner = new Scanner(System.in)) {
            System.out.print("Nombre del aprendiz: ");
            String nombre = scanner.nextLine().trim();

            // printf permite construir una salida formateada sin concatenaciones innecesarias.
            System.out.printf("Bienvenido, %s. El entorno Java funciona correctamente.%n", nombre);
        }
    }
}

File: 02-programacion-basica/01-entorno-sintaxis-y-entrada-salida/scala/ejemplos-guiados/HolaEntradaSalida.scala
import scala.io.StdIn.readLine

object HolaEntradaSalida:
  def main(args: Array[String]): Unit =
    // readLine obtiene texto desde la entrada estándar.
    val nombre = readLine("Nombre del aprendiz: ").trim
    println(s"Bienvenido, $nombre. El entorno Scala funciona correctamente.")
